modules/standard/IO.chpl

```diff
@@ -3168,35 +3182,49 @@ private inline proc _write_one_internal(_channel_internal:qio_channel_ptr_t,
 }
 
 pragma "no doc"
-proc channel.readIt(ref x) {
+proc channel.readIt(ref x) throws {
   if writing then compilerError("read on write-only channel");
   const origLocale = this.getLocaleOfIoRequest();
+  var error:syserr = ENOERR;
+
   on this.home {
     try! this.lock();
-    var error:syserr;
-    error = qio_channel_error(_channel_internal);
-    if ! error {
-      error = _read_one_internal(_channel_internal, kind, x, origLocale);
-      _qio_channel_set_error_unlocked(_channel_internal, error);
+    var etmp:syserr;
+    etmp = qio_channel_error(_channel_internal);
+    if !etmp {
+      etmp = _read_one_internal(_channel_internal, kind, x, origLocale);
+      _qio_channel_set_error_unlocked(_channel_internal, etmp);
     }
+    error = etmp;
     this.unlock();
   }
+
+  if !error then
+    try this._ch_ioerror(error, "in channel.readIt(" +
+                                _args_to_proto(x, preArg="ref ") + ")");
 }
 
 pragma "no doc"
-proc channel.writeIt(const x) {
+proc channel.writeIt(const x) throws {
   if !writing then compilerError("write on read-only channel");
   const origLocale = this.getLocaleOfIoRequest();
+  var error:syserr = ENOERR;
+
   on this.home {
     try! this.lock();
-    var error:syserr;
-    error = qio_channel_error(_channel_internal);
-    if ! error {
-      error = _write_one_internal(_channel_internal, kind, x, origLocale);
-      _qio_channel_set_error_unlocked(_channel_internal, error);
+    var etmp:syserr;
+    etmp = qio_channel_error(_channel_internal);
+    if !etmp {
+      etmp = _write_one_internal(_channel_internal, kind, x, origLocale);
+      _qio_channel_set_error_unlocked(_channel_internal, etmp);
     }
+    error = etmp;
     this.unlock();
   }
+
+  if !error then
```

Comment:
I think this should be "if error"?

Reply:
Mistake on my part - forgot that `QIO_ERROR_NONE = 0`. Logically, this statement is saying "if there was an error, then...".